src/gwc/src/main/java/org/geoserver/gwc/controller/GwcWmtsRestUrlHandlerMapping.java

```diff
@@ -21,10 +22,14 @@
  * Specific URL mapping handler for GWC WMTS REST API. The main goal of this handler id to handle
  * virtual services, it makes sure URLs with an workspace are correctly mapped and that a local
  * workspace is set and removed when needed.
+ *
+ * <p>SUBCLASSES: Set the `handlerMappingString` in the constructor.
  */
-public final class GwcWmtsRestUrlHandlerMapping extends RequestMappingHandlerMapping
+public class GwcWmtsRestUrlHandlerMapping extends RequestMappingHandlerMapping
```

Comment:
Maybe this should go in gwc-rest then?

Reply:
This cannot move as it is being subclassed, however the subclassing is mostly providing a single string?

Casual suggestion:  Perhaps leave this here and use an argument to configure the instances in other modules?

Reply:
* [ ] change to GwcWmtsRestUrlHandlerMapping to GwcUrlHandlerMapping
* [ ] add parameter to GwcUrlHandlerMapping to accept the pattern string
* [ ] refactor and remove the 3 real classes
* [ ] replace with refs to GwcUrlHandlerMapping (with pattern constructor) in application contexts